warp10/src/main/java/io/warp10/script/WarpScriptMacroLibrary.java

```diff
@@ -46,8 +49,25 @@
  * TODO(hbs): add support for secure script (the keystore is not initialized)
  */
 public class WarpScriptMacroLibrary {
-  private static final Map<String,Macro> macros = new HashMap<String, Macro>();
+  private static final Map<String,Macro> macros;
   
+  private static final int DEFAULT_CACHE_SIZE = 10000;
+  
+  static {
+    //
+    // Create macro map
+    //
+    
+    final int maxcachesize = Integer.parseInt(WarpConfig.getProperty(Configuration.WARPSCRIPT_LIBRARY_CACHE_SIZE, Integer.toString(DEFAULT_CACHE_SIZE)));
+    
+    macros = new LinkedHashMap<String,Macro>() {
+      @Override
+      protected boolean removeEldestEntry(java.util.Map.Entry<String,Macro> eldest) {
+        return this.size() > maxcachesize;
+      }
+    };
+
```

Comment:
What about a warning somewhere in the logs to tell user that some macro are not in the cache ?

Reply:
This warning would be written every time a macro would be loaded once you reach the limit. Setting a Sensision metric with the number of cached macros is probably a better option.

Reply:
Indeed. With 10000 as default, I doubt this will happen too often.

Reply:
Upated.